containers/base/Dockerfile

```diff
@@ -1,252 +1,293 @@
+    pip install --upgrade-strategy only-if-needed /datalab/lib/pydatalab/solutionbox/structured_data/. && \
+    pip3 install --upgrade-strategy only-if-needed --no-cache-dir . && \
+    pip3 install --upgrade-strategy only-if-needed /datalab/lib/pydatalab/solutionbox/image_classification/. && \
+    pip3 install --upgrade-strategy only-if-needed /datalab/lib/pydatalab/solutionbox/structured_data/. && \
+    jupyter nbextension install --py datalab.notebook && \
+    jupyter nbextension install --py google.datalab.notebook && \
+    jupyter nbextension enable --py widgetsnbextension && \
+    rm datalab/notebook/static/*.js google/datalab/notebook/static/*.js && \
+    mkdir -p /datalab/nbconvert && \
+    cp -R /usr/local/share/jupyter/nbextensions/gcpdatalab/* /datalab/nbconvert && \
+    ln -s /usr/local/lib/python2.7/dist-packages/notebook/static/custom/custom.css /datalab/nbconvert/custom.css && \
+    cd /
+
+## Use Debian unstable via pinning -- new style via APT::Default-Release
+RUN echo "deb http://http.debian.net/debian sid main" > /etc/apt/sources.list.d/debian-unstable.list \
+	&& echo 'APT::Default-Release "testing";' > /etc/apt/apt.conf.d/default
+
+ENV R_BASE_VERSION 3.4.3
+
+## Now install R and littler, and create a link for littler in /usr/local/bin
+## Also set a default CRAN repo, and make sure littler knows about it too
+RUN apt-get update \
+	&& apt-get install -t unstable -y --no-install-recommends \
+		libssh2-1-dev \
+		libcurl4-openssl-dev \
+		libssl-dev \
+		littler \
+                r-cran-littler \
+		r-base=${R_BASE_VERSION}* \
+		r-base-dev=${R_BASE_VERSION}* \
+		r-recommended=${R_BASE_VERSION}* \
+        && echo 'options(repos = c(CRAN = "https://cran.rstudio.com/"), download.file.method = "libcurl")' >> /etc/R/Rprofile.site \
+        && echo 'source("/etc/R/Rprofile.site")' >> /etc/littler.r \
+	&& ln -s /usr/share/doc/littler/examples/install.r /usr/local/bin/install.r \
+	&& ln -s /usr/share/doc/littler/examples/install2.r /usr/local/bin/install2.r \
+	&& ln -s /usr/share/doc/littler/examples/installGithub.r /usr/local/bin/installGithub.r \
+	&& ln -s /usr/share/doc/littler/examples/testInstalled.r /usr/local/bin/testInstalled.r \
+	&& install.r docopt \
+	&& rm -rf /tmp/downloaded_packages/ /tmp/*.rds \
+&& rm -rf /var/lib/apt/lists/*
+
+RUN 	R -e 'install.packages("devtools")' && \
+	R -e 'devtools::install_github("IRkernel/IRkernel")' && \
+	# or 'devtools::install_local("IRkernel-master.tar.gz")' && \
+	R -e 'IRkernel::installspec()' && \
+# to register the kernel in the current R installation 
+	R -e 'install.packages("googleAuthR"); library(googleAuthR); gar_gce_auth()' && \
+	R -e 'install.packages("bigQueryR"); install.packages("googleCloudStorageR")' && \
+	R -e 'install.packages("feather")' && \
+	R -e 'install.packages("tensorflow")' && \
+	R -e 'devtools::install_github("apache/spark@v2.2.0", subdir="R/pkg")'
+	
```

Comment:
New line at the end of the file.

Reply:
Author of `googleAuthR/googleCloudStorageR/bigQueryR` here - cool these can be used :)  I would also however suggest that `bigrquery` is installed as well as that is the more popular R BigQuery client, having deeper integration with popular tools like the `tidyverse`.  `bigQueryR` does carry features such as uploading and authentication that will be useful though, I imagine it would be best for R users to have the option of both. e.g add `install.packages("bigrquery")`, or if following above suggestion on using `install2.r`:

```
RUN install2.r --error \ 
    -r 'http://cran.rstudio.com' \
    googleAuthR \ 
    googleCloudStorageR \
    bigQueryR \ 
    bigrquery \
    feather \
    tensorflow 
RUN R -e 'gar_sce_auth()'
```

Reply:
I would actually start from [Rocker/tidyverse](https://hub.docker.com/r/rocker/tidyverse/) Dockefile to build this one since most R users nowadays work vastly with these packages.